pkg/detector/detector_test.go

```diff
@@ -0,0 +1,301 @@
+package detector
+
+import (
+	"testing"
+
+	appsv1 "k8s.io/api/apps/v1"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+
+	workloadv1alpha1 "github.com/karmada-io/karmada/examples/customresourceinterpreter/apis/workload/v1alpha1"
+	"github.com/karmada-io/karmada/pkg/util"
+)
+
+type MockWorker struct {
+	EnqueueCount int
+	util.AsyncWorker
+}
+
+func (m *MockWorker) Add(item interface{}) {
+	m.EnqueueCount++
+}
+
+func (m *MockWorker) Enqueue(obj runtime.Object) {
+	m.Add(nil)
+}
+
+func (m *MockWorker) Run(workerNumber int, stopChan <-chan struct{}) {
+
+}
+
+func Ptr[T any](v T) *T {
+	return &v
+}
+
+func TestResourceDetector_OnUpdate(t *testing.T) {
+	tests := []struct {
+		name    string
+		oldObj  interface{}
+		newObj  interface{}
+		wantAdd bool
+	}{
+		{
+			name: "Only user defined fields changed(Deployment)",
+			oldObj: &appsv1.Deployment{
+				TypeMeta: metav1.TypeMeta{Kind: "Deployment", APIVersion: "v1"},
```

Comment:
A little advise.  We have declared the struct type. I think we can omit the `TypeMeta` field.

Reply:
> A little advise. We have declared the struct type. I think we can omit the `TypeMeta` field.

Thanks for point out, it's modified in my new commit